lib/src/components/snack_bar/snack_bar.dart

```diff
@@ -0,0 +1,349 @@
+    return Padding(
+      padding: const EdgeInsets.symmetric(vertical: ZetaSpacing.xxs),
+      child: IconButton(
+        style: IconButton.styleFrom(
+          padding: const EdgeInsets.symmetric(horizontal: ZetaSpacing.s),
+          tapTargetSize: MaterialTapTargetSize.shrinkWrap,
+          minimumSize: const Size(20, 20),
+        ),
+        onPressed: onPressed,
+        icon: Icon(
+          ZetaIcons.close_round,
+          color: color,
+          size: 20,
+        ),
+      ),
+    );
+  }
+}
+
+class _ActionButton extends StatelessWidget {
+  const _ActionButton({
+    required this.onPressed,
+    required this.label,
+    required this.color,
+  });
+
+  final Color color;
+  final String? label;
+  final VoidCallback? onPressed;
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(ObjectFlagProperty<VoidCallback?>.has('onPressed', onPressed))
+      ..add(StringProperty('label', label))
+      ..add(ColorProperty('color', color));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: ZetaSpacing.s),
+      child: TextButton(
+        style: TextButton.styleFrom(
+          textStyle: ZetaTextStyles.labelLarge,
+          padding: const EdgeInsets.symmetric(
+            horizontal: ZetaSpacing.s,
+            vertical: ZetaSpacing.xxs,
+          ),
+          tapTargetSize: MaterialTapTargetSize.shrinkWrap,
+          minimumSize: Size.zero,
+        ),
+        onPressed: onPressed,
+        child: Text(
+          label ?? '',
+          style: ZetaTextStyles.labelLarge.copyWith(color: color),
+        ),
+      ),
+    );
+  }
+}
+
+class _LeadingIcon extends StatelessWidget {
+  const _LeadingIcon(this.type, this.icon);
+
+  final Icon? icon;
+  final ZetaSnackBarType? type;
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(EnumProperty<ZetaSnackBarType?>('type', type));
+  }
+
+  Color _getIconColor(ZetaColors colors, ZetaSnackBarType? type) {
+    return switch (type) {
+      ZetaSnackBarType.positive => colors.positive,
+      ZetaSnackBarType.info => colors.info,
+      ZetaSnackBarType.warning => colors.warning,
+      ZetaSnackBarType.error || ZetaSnackBarType.deletion => colors.negative,
+      ZetaSnackBarType.view => colors.primary,
+      _ => colors.iconInverse,
+    };
+  }
+
+  Widget _getIcon(ZetaSnackBarType? type) {
+    return switch (type) {
+      ZetaSnackBarType.positive => const Icon(ZetaIcons.check_circle_round),
+      ZetaSnackBarType.info => const Icon(ZetaIcons.info_round),
+      ZetaSnackBarType.warning => const Icon(ZetaIcons.warning_round),
+      ZetaSnackBarType.error => const Icon(ZetaIcons.error_round),
+      ZetaSnackBarType.deletion => const Icon(ZetaIcons.delete_round),
+      _ => const SizedBox(),
```

Comment:
I was wrong, there is an icon for ZetaSnackbarType.view:
`ZetaIcons.open_in_new_window_round` 
![image](https://github.com/zebrafed/zeta_flutter/assets/23633655/22f269e2-538c-4842-9fd4-0300f677311f)